content/silver/dev/emacs.md

```diff
@@ -0,0 +1,211 @@
+---
+title: Emacs Support with LSP
+weight: 250
+---
+
+{{< toc >}}
+
+
+Emacs LSP support allows for copper specific syntax highlighting through semantic tokens, lookup of declarations (preliminary), and showing errors and warnings in silver buffers. 
+
+## Prerequisites
+Above and beyond the standard Silver requirements, you need Maven (to build the silver lsp server), and of course, Emacs (26+ recommended).
+
+## Configuring Emacs prerequisites
+
+To get started, you will need to have the package [lsp-mode](https://emacs-lsp.github.io/lsp-mode/page/installation/)  installed. Note that some features of lsp-mode require additional packages, but configuring lsp-mode is a
+bit out of scope of this article. There's more information about the additional features in [here](https://emacs-lsp.github.io/lsp-mode/page/installation/#vanilla-emacs), if you're configuring with vanilla emacs.
+
+Once you have that, you'll also need to have the [silver-mode.el](https://github.com/melt-umn/silver/blob/develop/support/emacs/silver-mode/silver-mode.el)
+file loaded in some manner. This can be done by adding the path to the file to your `load-path`.
+
+``` elisp
+(add-to-list 'load-path "/path/to/silver/support/emacs/silver-mode/silver-mode.el")
+```
+
+But if you do use a package manager like [straight](https://github.com/radian-software/straight.el), I made a recipe for setting up the silver-mode as a package.
+
+```elisp
+(straight-use-package
+  `(silver-mode :type git :repo "melt-umn/silver" :files ("support/emacs/silver-mode/*.el")))
+```
+
+
+
+## Setting up LSP
+
+`LSP` has two components, the client and the server.
+
+### Setting up LSP Server 
+
+Checkout [copper](https://github.com/melt-umn/copper) somewhere on your system, and run `mvn install` in that repo to build and install copper.
+
+Then, go to `silver/language-server/` and run `./build.sh` to build the language server. When it finishes
+you should have a jar `/silver/language-server/launcher/target/launcher.jar`.
+
+After that, make a script `silver-language-server`, put it in your `$PATH` somehow (I put it in `~/bin/`, where the other silver scripts are linked to), and put the following contents into it:
+
+```sh
+#!/usr/bin/env sh
+
+java -Xmx7G -Xss21M -jar /path/to/silver/language-server/launcher/target/launcher.jar
+
+```
+
+This lets us launch the server by calling `silver-language-server` in our shell.
+
+### Setting up Emacs as a Silver LSP client
+
+This is where advice becomes fragmented, as the ways and methods of configuring emacs vary. I have two
+methods outlined here, one for vanilla emacs users, and one for users of [Doom Emacs](https://github.com/doomemacs/doomemacs/).
+
+
+#### Vanilla Emacs
```

Comment:
Probably better to check these in somewhere under `support/emacs` and link to them here rather than having people copy/paste from here.